astropy/__init__.py

```diff
@@ -189,6 +189,9 @@ def test(package=None, test_path=None, args=None, plugins=None,
     skip_docs : bool, optional
         When `True`, skips running the doctests in the .rst files.
 
+    kselect : str, optional
+        Test selection, passed to py.test as "-k <kselect>".
```

Comment:
`-k` is fine for the short name.   In principle we don't even need a long name, but if we had one I'd just say `--test-name` or something like that.  The point is that it matches the test name against a pattern.